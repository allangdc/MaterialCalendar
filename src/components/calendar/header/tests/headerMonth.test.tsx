import React from "react";
import Calendar from "../..";
import { cleanup, fireEvent, render, screen } from "@testing-library/react";
import { advanceTo, clear } from "jest-date-mock";
import renderer from "react-test-renderer";

beforeEach(() => {
  render(<Calendar language={"pt"} />);
});

beforeAll(() => {
  advanceTo(new Date(1984, 5, 13));
});

afterAll(() => {
  clear();
});

test("Months must start in current date [current: 1984-06-13]", () => {
  const monthTitle = screen.getByTestId("HeaderTitle");
  expect(monthTitle).toHaveTextContent("Junho 1984");
});

test("Months must decrement when clicked on back button", () => {
  const monthTitle = screen.getByTestId("HeaderTitle");
  const backBtn = screen.getByTestId("btn-back");
  fireEvent.click(backBtn);
  expect(monthTitle).toHaveTextContent("Maio 1984");
});

test("Months must increment when clicked on forward button", () => {
  const monthTitle = screen.getByTestId("HeaderTitle");
  const btn = screen.getByTestId("btn-forward");
  fireEvent.click(btn);
  expect(monthTitle).toHaveTextContent("Julho 1984");
});

test("Today button must change currentDate for Today value", () => {
  const monthTitle = screen.getByTestId("HeaderTitle");
  const btn = screen.getByTestId("btn-forward");
  const btnToday = screen.getByTestId("btn-today");
  // Jump to January 2000
  for (let i = 0; i < 12 * 16 - 5; i++) {
    fireEvent.click(btn);
  }
  expect(monthTitle).toHaveTextContent("Janeiro 2000");
  fireEvent.click(btnToday);
  expect(monthTitle).toHaveTextContent("Junho 1984");
});

test("Snapshot of the Header in monthly view", () => {
  cleanup();
  const component = renderer.create(<Calendar language={"pt"} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
