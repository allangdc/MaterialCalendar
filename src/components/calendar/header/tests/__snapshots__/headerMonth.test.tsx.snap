// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of the Header in monthly view 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1em9ut6-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
      id="mobile-header"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1em9ut6-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined css-11lq3yg-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true makeStyles-header_back_button-23 css-4jz4pb-MuiGrid-root"
          >
            <button
              aria-label="back the date for calrndar"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall css-2rshpx-MuiButtonBase-root-MuiIconButton-root"
              data-testid="btn-back"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowBackIosNewIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17.77 3.77 16 2 6 12l10 10 1.77-1.77L9.54 12z"
                />
              </svg>
            </button>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 css-klo4dh-MuiGrid-root"
          >
            <h6
              className="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter css-1kgxqm0-MuiTypography-root"
              data-testid="HeaderTitle"
            />
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true makeStyles-header_foward_button-24 css-4jz4pb-MuiGrid-root"
          >
            <button
              aria-label="forward the date for calendar"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall css-2rshpx-MuiButtonBase-root-MuiIconButton-root"
              data-testid="btn-forward"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowForwardIosIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-4jz4pb-MuiGrid-root"
      >
        <button
          className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
          data-testid="btn-today"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "height": 30,
            }
          }
          tabIndex={0}
          type="button"
        >
          Hoje
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-4jz4pb-MuiGrid-root"
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "flex-end",
            }
          }
        >
          <div
            className="MuiToggleButtonGroup-root css-1gjgmky-MuiToggleButtonGroup-root"
            role="group"
            style={
              Object {
                "height": 30,
              }
            }
          >
            <button
              aria-pressed={true}
              className="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-ueukts-MuiButtonBase-root-MuiToggleButton-root"
              data-testid="toggle-month"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
              value={0}
            >
              MÃªs
            </button>
            <button
              aria-pressed={false}
              className="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-ueukts-MuiButtonBase-root-MuiToggleButton-root"
              data-testid="toggle-year"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
              value={1}
            >
              Ano
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1em9ut6-MuiGrid-root"
  />
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1em9ut6-MuiGrid-root"
  >
    <div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined css-11lq3yg-MuiGrid-root"
        id="monthly"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1em9ut6-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined css-11lq3yg-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-4jz4pb-MuiGrid-root"
            >
              <div
                className="makeStyles-headercell-21"
                style={
                  Object {
                    "borderColor": "rgb(223 223 222)",
                  }
                }
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1uwgr7b-MuiTypography-root"
                >
                  D
                </p>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-4jz4pb-MuiGrid-root"
            >
              <div
                className="makeStyles-headercell-21"
                style={
                  Object {
                    "borderColor": "rgb(223 223 222)",
                  }
                }
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1uwgr7b-MuiTypography-root"
                >
                  S
                </p>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-4jz4pb-MuiGrid-root"
            >
              <div
                className="makeStyles-headercell-21"
                style={
                  Object {
                    "borderColor": "rgb(223 223 222)",
                  }
                }
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1uwgr7b-MuiTypography-root"
                >
                  T
                </p>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-4jz4pb-MuiGrid-root"
            >
              <div
                className="makeStyles-headercell-21"
                style={
                  Object {
                    "borderColor": "rgb(223 223 222)",
                  }
                }
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1uwgr7b-MuiTypography-root"
                >
                  Q
                </p>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-4jz4pb-MuiGrid-root"
            >
              <div
                className="makeStyles-headercell-21"
                style={
                  Object {
                    "borderColor": "rgb(223 223 222)",
                  }
                }
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1uwgr7b-MuiTypography-root"
                >
                  Q
                </p>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-4jz4pb-MuiGrid-root"
            >
              <div
                className="makeStyles-headercell-21"
                style={
                  Object {
                    "borderColor": "rgb(223 223 222)",
                  }
                }
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1uwgr7b-MuiTypography-root"
                >
                  S
                </p>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-4jz4pb-MuiGrid-root"
            >
              <div
                className="makeStyles-headercell-21"
                style={
                  Object {
                    "borderColor": "rgb(223 223 222)",
                  }
                }
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1uwgr7b-MuiTypography-root"
                >
                  S
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1em9ut6-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined css-11lq3yg-MuiGrid-root"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
